rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -329,100 +287,127 @@ frontend httpfront-{{ $host }}
+    use_backend error413 if {{ $server.ACLLabel }} { var(txn.path) -m beg {{ $location.Path }} } { req.body_size gt {{ sizeSuffix $location.Proxy.BodySize }} }
+{{- end }}
 {{- end }}
 {{- end }}
 
-{{- if $sslconn }}
+{{- /*------------------------------------*/}}
+{{- range $server := $servers }}
 {{- if $server.HSTS }}
 {{- $hsts := $server.HSTS }}
 {{- if $hsts.Enable }}
-    http-response set-header Strict-Transport-Security "max-age={{ $hsts.MaxAge }}{{ if $hsts.Subdomains }}; includeSubDomains{{ end }}{{ if $hsts.Preload }}; preload{{ end }}" if from-https
+    http-response set-header Strict-Transport-Security "max-age={{ $hsts.MaxAge }}{{ if $hsts.Subdomains }}; includeSubDomains{{ end }}{{ if $hsts.Preload }}; preload{{ end }}" if from-https {{ $server.ACLLabel }}
 {{- end }}
 {{- else }}
 {{- range $location := $server.Locations }}
 {{- $hsts := $location.HSTS }}
 {{- if $hsts.Enable }}
-    http-response set-header Strict-Transport-Security "max-age={{ $hsts.MaxAge }}{{ if $hsts.Subdomains }}; includeSubDomains{{ end }}{{ if $hsts.Preload }}; preload{{ end }}" if from-https{{ $location.HAMatchTxnPath }}
+    http-response set-header Strict-Transport-Security "max-age={{ $hsts.MaxAge }}{{ if $hsts.Subdomains }}; includeSubDomains{{ end }}{{ if $hsts.Preload }}; preload{{ end }}" if from-https {{ $server.ACLLabel }}{{ $location.HAMatchTxnPath }}
 {{- end }}
 {{- end }}
 {{- end }}
 {{- end }}
 
+{{- /*------------------------------------*/}}
+{{- range $server := $servers }}
 {{- range $location := $server.Locations }}
 {{- if not $location.IsRootLocation }}
-    use_backend {{ $location.Backend }} if { var(txn.path) -m beg {{ $location.Path }} }
+    use_backend {{ $location.Backend }} if {{ $server.ACLLabel }} { var(txn.path) -m beg {{ $location.Path }} }
+{{- else if $server.ACLLabel }}
+    use_backend {{ $location.Backend }} if {{ $server.ACLLabel }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- if $isShared }}
+    default_backend httpback-default-backend
+{{- else if $isDefault }}
+    default_backend {{ $ing.DefaultServer.RootLocation.Backend }}
+{{- else if $isCACert }}
+    default_backend httpback-shared-backend
+{{- end }}
+{{- end }}{{/* define "http_front" */}}
+
+{{- /*------------------------------------*/}}
+{{- /*------------------------------------*/}}
+{{- define "acl" }}
+{{- $cfg := .p1 }}
+{{- $server := .p2 }}
+{{- $fetch := .p3 }}
+{{- $needport := .p4 }}
+{{- if ne $server.ACLLabel "" }}
+{{- if isWildcardHostname $server.Hostname }}
+    acl {{ $server.ACLLabel }} {{ $fetch }} -m reg -i {{ hostnameRegex $server.Hostname }}
 {{- else }}
-    default_backend {{ $location.Backend }}
+    acl {{ $server.ACLLabel }} {{ $fetch }} -i {{ $server.Hostname }}{{ if $needport }} {{ $server.Hostname }}:80{{ if and $cfg.HTTPStoHTTPPort (ne $cfg.HTTPStoHTTPPort 80) }} {{ $server.Hostname }}:{{ $cfg.HTTPStoHTTPPort }}{{ end }}{{ end }}
```

Comment:
Testing looks good so far except now `<host>:443` needs to be matched just like `<host>:80`, because some clients set the `Host` header to include the port. This is something I had to fix in [PR #140](https://github.com/jcmoraisjr/haproxy-ingress/pull/124/commits/ea2216295fcfd2fe7b83c1dd13a32ff6e1baa20b#diff-0aee42afd25a32b87c1c27ad8c705c40R143) as well. So something like the following:

```
acl {{ $server.ACLLabel }} {{ $fetch }} -i {{ $server.Hostname }}{{ if $needport }} {{ $server.Hostname }}:80 {{ $server.Hostname }}:443{{ if and $cfg.HTTPStoHTTPPort (ne $cfg.HTTPStoHTTPPort 80) }} {{ $server.Hostname }}:{{ $cfg.HTTPStoHTTPPort }}{{ end }}{{ end }}
```

Reply:
Makes sense since ssl requests now matches host header. Updated.